@remirror/core/src/helpers/utils.ts

```diff
@@ -185,6 +189,44 @@ export const findParentNode = ({
   return;
 };
 
+/**
+ * Finds the node at the passed selection.
+ */
+export const findNodeAtSelection = (selection: Selection): FindParentNodeResult => {
+  return findParentNode({ predicate: () => true, selection })!;
+};
+
+/**
+ * Finds the node at the end of the Prosemirror document.
+ *
+ * @param doc - the parent doc node of the editor which contains all the other nodes.
+ */
+export const findNodeAtEndOfDoc = (doc: ProsemirrorNode) => findNodeAtPosition(PMSelection.atEnd(doc).$from);
+
+/**
+ * Finds the node at the start of the prosemirror.
+ *
+ * @param doc - the parent doc node of the editor which contains all the other nodes.
+ */
+export const findNodeAtStartOfDoc = (doc: ProsemirrorNode) =>
+  findNodeAtPosition(PMSelection.atStart(doc).$from);
+
+/**
+ * Finds the node at the resolved position.
+ *
+ * @param $pos - the resolve position in the document
+ */
+export const findNodeAtPosition = ($pos: ResolvedPos): FindParentNodeResult => {
```

Comment:
This helper might resolve my question in 'Proposed Implementation' in https://github.com/ifiokjr/remirror/issues/111.

Reply:
I'll take a look